contrib/hamilton/contrib/user/zilto/nixtla_statsforecast/__init__.py

```diff
@@ -0,0 +1,203 @@
+import logging
+from typing import Callable, Optional
+
+logger = logging.getLogger(__name__)
+
+from hamilton import contrib
+
+with contrib.catch_import_errors(__name__, __file__, logger):
+    import matplotlib
+    import pandas as pd
+    from statsforecast import StatsForecast
+    from statsforecast.models import (
+        _TS,
+        AutoARIMA,
+        CrostonClassic,
+        DynamicOptimizedTheta,
+        HistoricAverage,
+        HoltWinters,
+        SeasonalNaive,
+    )
+    from utilsforecast.evaluation import evaluate
+    from utilsforecast.losses import mse
+
+
+def raw_dataset() -> pd.DataFrame:
+    """Download Nixtla's version of the M4 hourly dataset. ref: https://paperswithcode.com/dataset/m4
+
+    Needs to have columns an `id`, a `time`, and a `target` columns.
+    Nixtla defaults are:
+        id: "unique_id"
+        time: "ds"
+        target: "y"
+
+    """
+    return pd.read_parquet("https://datasets-nixtla.s3.amazonaws.com/m4-hourly.parquet")
+
+
+def dataset(raw_dataset: pd.DataFrame, n_ids: int = 5, n_days: int = 7) -> pd.DataFrame:
```

Comment:
Names `n_ids`, `n_days` should probably be more descriptive IMO. E.G. how are they used? They'll becomes inputs to the DAG.